libcron/include/libcron/SortableQueue.h

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <vector>
+#include <map>
+#include <unordered_map>
+#include <vector>
+#include "Task.h"
+
+namespace libcron
+{           
+    template<typename LockType>
+    class SortableQueue
```

Comment:
SortableQueue is a generic name, this tailored for Tasks, no lets name it TaskQueue (and rename the file too)